src/main/kotlin/org/rust/ide/console/RsConsoleView.kt

```diff
@@ -0,0 +1,105 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.console
+
+import com.intellij.execution.console.LanguageConsoleImpl
+import com.intellij.execution.impl.ConsoleViewUtil
+import com.intellij.execution.process.ProcessOutputTypes
+import com.intellij.execution.ui.ConsoleViewContentType
+import com.intellij.execution.ui.ObservableConsoleView
+import com.intellij.openapi.editor.colors.EditorColors
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.ActionCallback
+import com.intellij.openapi.util.Key
+import com.intellij.openapi.wm.IdeFocusManager
+import com.intellij.psi.util.PsiTreeUtil
+import org.rust.cargo.project.model.cargoProjects
+import org.rust.lang.RsLanguage
+import org.rust.lang.core.psi.*
+import org.rust.openapiext.toPsiFile
+import javax.swing.JComponent
+
+class RsConsoleView(project: Project, title: String, private val consoleRunner: RsConsoleRunnerImpl)
+    : LanguageConsoleImpl(project, title, RsLanguage), ObservableConsoleView {
+
+    private lateinit var myExecuteActionHandler: RsConsoleExecuteActionHandler
+
+    private val myInitialized: ActionCallback = ActionCallback()
+    private val codeFragment: RsReplCodeFragment? = virtualFile.toPsiFile(project) as? RsReplCodeFragment
+
+    init {
+        val virtualFile = virtualFile
+        virtualFile.putUserData(RUST_CONSOLE_KEY, true)
+        // Mark editor as console one, to prevent autopopup completion
+        historyViewer.putUserData(ConsoleViewUtil.EDITOR_IS_CONSOLE_HISTORY_VIEW, true)
+        super.setPrompt(">> ")
+        setUpdateFoldingsEnabled(false)
+    }
+
+    fun setExecutionHandler(consoleExecuteActionHandler: RsConsoleExecuteActionHandler) {
```

Comment:
Do you really need this setter? Or maybe just use the property?

Reply:
Yes, property is enough